dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js

```diff
@@ -0,0 +1,28 @@
+const db = require('../../../../drivers/db-client');
+
+const getUtilisationReports = async (req, res) => {
+  const { bankId } = req.params;
+
+  const utilisationReportsCollection = await db.getCollection('utilisation-reports');
+  const filteredAndSortedUtilisationReports = await utilisationReportsCollection.aggregate([
+    {
+      $match: {
+        bankId,
+      },
+    },
+    {
+      $sort: {
+        year: 1,
+        month: 1,
+      },
+    },
+  ]).toArray();
+  if (filteredAndSortedUtilisationReports) {
+    return res.status(200).send(filteredAndSortedUtilisationReports);
+  }
+  return res.status(404).send({ status: 404, message: 'No utilisation reports found' });
```

Comment:
I'm not sure 404 is right here, 404 to me is we've hit a route that doesn't exist.
If there are no reports isfilteredAndSortedUtilisationReports `[]` or `null` or `undefined`? If [] then that will be truthy in javascript and we'll return the 200 and empty array which is fine.
If it's not [] then we can assume an error happened (do we need a try/catch?) and we can send back 500 and Error finding reports?

Reply:
Yes yes, good point!